build.sbt

```diff
@@ -0,0 +1,535 @@
+  scalaVersion := bootstrapScalaVersion,
+  // we don't cross build Scala itself
+  crossPaths := false,
+  // do not add Scala library jar as a dependency automatically
+  autoScalaLibrary := false,
+  // we also do not add scala instance automatically because it introduces
+  // a circular instance, see: https://github.com/sbt/sbt/issues/1872
+  managedScalaInstance := false,
+  // this is a way to workaround issue described in https://github.com/sbt/sbt/issues/1872
+  // check it out for more details
+  scalaInstance := ScalaInstance(scalaVersion.value, appConfiguration.value.provider.scalaProvider.launcher getScala scalaVersion.value),
+  // we always assume that Java classes are standalone and do not have any dependency
+  // on Scala classes
+  compileOrder := CompileOrder.JavaThenScala,
+  javacOptions in Compile ++= Seq("-g", "-source", "1.5", "-target", "1.6"),
+  // we don't want any unmanaged jars; as a reminder: unmanaged jar is a jar stored
+  // directly on the file system and it's not resolved through Ivy
+  // Ant's build stored unmanaged jars in `lib/` directory
+  unmanagedJars in Compile := Seq.empty,
+  sourceDirectory in Compile := baseDirectory.value,
+  unmanagedSourceDirectories in Compile := List(baseDirectory.value),
+  scalaSource in Compile := (sourceDirectory in Compile).value,
+  javaSource in Compile := (sourceDirectory in Compile).value,
+  // resources are stored along source files in our current layout
+  resourceDirectory in Compile := (sourceDirectory in Compile).value,
+  // each subproject has to ask specifically for files they want to include
+  includeFilter in unmanagedResources in Compile := NothingFilter,
+  target := (baseDirectory in ThisBuild).value / "target" / thisProject.value.id,
+  target in Compile in doc := buildDirectory.value / "scaladoc" / thisProject.value.id,
+  classDirectory in Compile := buildDirectory.value / "quick/classes" / thisProject.value.id,
+  // given that classDirectory is overriden to be _outside_ of target directory, we have
+  // to make sure its being cleaned properly
+  cleanFiles += (classDirectory in Compile).value,
+  fork in run := true,
+  credentialsSetting,
+  publishSetting,
+  pomExtra := (
+    <url>https://github.com/typelevel/scala</url>
+    <licenses>
+      <license>
+        <name>Apache License</name>
+        <url>http://opensource.org/licenses/Apache-2.0</url>
+        <distribution>repo</distribution>
+      </license>
+    </licenses>
+    <scm>
+      <url>git@github.com:typelevel/scala.git</url>
+      <connection>scm:git:git@github.com:typelevel/scala.git</connection>
+    </scm>
+    <developers>
+      {
+        Seq(
+          ("folone"     , "George Leontiev"),
+          ("larsrh"     , "Lars Hupel"),
+          ("milessabin" , "Miles Sabin"),
+          ("non"        , "Erik Osheim"),
+          ("stacycurl"  , "Stacy Curl"),
+          ("stew"       , "Mike O'Connor"),
+          ("tixxit"     , "Tom Switzer")
```

Comment:
I've only added people who are memebers of the github typelevel org. It would be nice if someone contributed an sbt task that would hook into `publish` and `publish-signed`, and populated this list with actual contributors.